File: samples/FsBulletML.Sample.Unity2D.CSharp/Assets/Scripts/BgScroll.cs
using UnityEngine;

public class BgScroll : MonoBehaviour
{
    [SerializeField]
    private float scrollSpeed1 = 0.1f;
    void Update()
    {
        var newTextureOffset = this.renderer.material.mainTextureOffset;
        newTextureOffset.y = this.renderer.material.mainTextureOffset.y - Time.deltaTime * scrollSpeed1;
        this.renderer.material.mainTextureOffset = newTextureOffset;
    }
}

File: samples/FsBulletML.Sample.Unity2D.FSharp/FsBulletML.Sample.Unity2D.FSharp/BgScroll.fs
namespace FsBulletML.Sample.Unity2D.FSharp

open System
open UnityEngine

type BgScroll () =
  inherit MonoBehaviour ()
  [<SerializeField;DefaultValue>]val mutable public scrollSpeed : float32
  
  member this.Update () =
    let newTextureOffset = new Vector2(this.renderer.material.mainTextureOffset.x , this.renderer.material.mainTextureOffset.y - Time.deltaTime * this.scrollSpeed)
    this.renderer.material.mainTextureOffset <- newTextureOffset
